### PR TITLE
User Comments

### LINKED ISSUE #50: Comments & User Feedback
Users will be able to comment on projects and provide feedback similar to an upvote system.

Comments: Slack/Stack Overflow model - one main layer, any replies are nested one layer below the main comment

Upvote system: Piazza/Medium model - make it seem less like an explicit upvote but rather more like "a user found this helpful"

Timeline:

- Start: 4/9
- End: 4/13

### PR DESCRIPTION
**Pull request by coconut750750**
Resolves #50

### CODE REVIEW

**Comment by arpanlaha** [comment]
Cat Gif

File: auth-server/src/api/register.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ router.post(
       .isLength({ min: 1 })
   ],
   handleAsyncErrors(async function(req, res) {
+    console.log(req.body);
```
Comment (yousefa00) [inline auth-server/src/api/register.js:28]: Dont forget to remove this log

File: auth-server/src/utils/jwtHelpers.js [hunk 2]
```diff
@@ -31,10 +31,20 @@ function shouldUpdateJWT(token, email, role) {
     if (String(userEmail) === String(email) && userRole === role) {
       return false;
     }
-    return false;
+    return true;
   } catch (err) {
     return err;
   }
+  // } catch (err) {
```
Comment (yousefa00) [inline auth-server/src/utils/jwtHelpers.js:38]: If you dont think you'll reference this commented code in the near future, go ahead and remove it to keep the codebase clean

File: client/components/CommentsSection.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          />
+        </Row>
+      );
+    }
+    return commentList;
+  };
+
+  const fetchComments = () => {
+    getComments(projectId).then(data => {
+      const { comments } = data.data;
```
Comment (yousefa00) [inline client/components/CommentsSection.jsx:39]: This is a bit nitpicky, but could we be more explicit with naming the parent here such that it doesnt have the same name `data` as the child?

File: client/components/Comment.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  const renderThread = thread => {
+    let threadList = [];
+    for (let comment of thread) {
+      threadList.push(
+        renderBlock(comment.authorName, comment.createdAt, comment.content)
+      );
+    }
+
+    return threadList;
+  };
```
Comment (arpanlaha) [inline client/components/Comment.jsx:35]: this might make more sense as a single map function

File: client/components/CommentsSection.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import Comment from "./Comment";
+import CommentEditor from "./CommentEditor";
+
+export default function CommentsSection(props) {
+  const { projectId } = props;
+  const [comments, setComments] = useState([]);
+
+  useEffect(() => {
+    fetchComments();
+  }, []);
```
Comment (arpanlaha) [inline client/components/CommentsSection.jsx:20]: don't forget this lint warning

File: client/components/Comment.jsx (hunk 4 above)
Comment (arpanlaha) [inline client/components/Comment.jsx:35]: this might make more sense as a single map function

File: client/components/CommentsSection.jsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        <Row>
+          <Comment
+            comment={comment}
+            postThread={content => postThread(content, index)}
+          />
+        </Row>
+      );
+    }
+    return commentList;
+  };
```
Comment (arpanlaha) [inline client/components/CommentsSection.jsx:35]: another place where map might make more sense

**Pull request merged**